src/main/scala/scorex/core/network/peer/PeerDatabaseImpl.scala

```diff
@@ -27,9 +27,9 @@ class PeerDatabaseImpl(settings: NetworkSettings,
     whitelistPersistence.put(address, updatedPeerInfo)
   }
 
-  override def blacklistPeer(address: InetSocketAddress): Unit = {
+  override def blacklistPeer(address: InetSocketAddress, time: NetworkTime.Time): Unit = {
```

Comment:
@ceilican We don't use the global `TimeProvider`. We explicitly pass the timestamp itself in `blacklistPeer`. So `PeerDatabaseImpl` does not know anything about `TimeProvider`.

Reply:
Oh, yes. I missed that. I had just seen that you were not passing NetworkTimeProvider as an argument to the constructor of `PeerDatabaseImpl` anymore, and then I wrongly assumed that you would be accessing `NetworkTime` globally.

Now I see better what you did, and I think it was a good change. I think it is a good idea to pass `time` explicitly here, and let `PeerManager`, instead of `PeerDatabaseImpl`, worry about time and the network time provider.

Reply:
Reasonable change